icinga2/service.go

```diff
@@ -6,22 +6,23 @@ import (
 )
 
 type Service struct {
-	Name               string  `json:"name,omitempty"`
-	DisplayName        string  `json:"display_name"`
-	HostName           string  `json:"host_name"`
-	CheckCommand       string  `json:"check_command"`
-	EnableActiveChecks bool    `json:"enable_active_checks"`
-	Notes              string  `json:"notes"`
-	NotesURL           string  `json:"notes_url"`
-	ActionURL          string  `json:"action_url"`
-	Vars               Vars    `json:"vars"`
-	Zone               string  `json:"zone,omitempty"`
-	CheckInterval      float64 `json:"check_interval"`
-	RetryInterval      float64 `json:"retry_interval"`
-	MaxCheckAttempts   float64 `json:"max_check_attempts"`
-	CheckPeriod        string  `json:"check_period,omitempty"`
-	State              float64 `json:"state,omitempty"`
-	LastStateChange    float64 `json:"last_state_change,omitempty"`
+	Name               string   `json:"name,omitempty"`
+	DisplayName        string   `json:"display_name"`
+	HostName           string   `json:"host_name"`
+	CheckCommand       string   `json:"check_command"`
+	EnableActiveChecks bool     `json:"enable_active_checks"`
+	Notes              string   `json:"notes"`
+	NotesURL           string   `json:"notes_url"`
+	ActionURL          string   `json:"action_url"`
+	Vars               Vars     `json:"-"`
```

Comment:
This currently breaks reading any variables when fetching services with `ListServices()` or `GetService()` (e.g. in Signalilo's service garbage collection), since field `Vars` will always be empty when unmarshalling the Icinga2 response with tag `json:"-"`. From my local tests, it's fine to keep this as

```suggestion
	Vars               Vars     `json:"vars"`
```

Since we have the custom marshalling which takes precedence over the `json` tag, we don't overwrite vars injected from a service template, even if we have json tag `json:"vars"`.